pkg/ring/lifecycler.go

```diff
@@ -972,6 +1001,7 @@ func (i *Lifecycler) updateConsul(ctx context.Context) error {
 			instanceDesc.Zone = i.Zone
 			instanceDesc.RegisteredTimestamp = i.getRegisteredAt().Unix()
 			ringDesc.Ingesters[i.ID] = instanceDesc
+			i.delegate.OnRingInstanceHeartbeat(i, ringDesc)
```

Comment:
We should move this line outside the else block. We should call `OnRingInstanceHeartbeat` no matter the instance exists in the ring or not